kore/src/Kore/Variables/Target.hs

```diff
@@ -158,3 +159,13 @@ instance
     unparse (NonTarget var) = unparse var
     unparse2 (Target var) = unparse2 var
     unparse2 (NonTarget var) = unparse2 var
+
+targetIfEqual
+    :: Eq variable
+    => variable
```

Comment:
I think this is only used on `ElementVariable`s anyway, and we should always try to keep the `ElementVariable` or `SetVariable` wrapper:

```suggestion
    => ElementVariable variable
```